packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -276,6 +276,19 @@ abstract class FlutterCommand extends Command<void> {
       help: 'Build a release version of your app${defaultToRelease ? ' (default mode)' : ''}.');
   }
 
+  void addShrinkingFlag() {
+    argParser.addFlag('shrinking',
+      negatable: true,
+      defaultsTo: true,
```

Comment:
My only non-nit for this: should we start by defaulting this to false to make it opt in for now? That would give people a chance to try this in the wild before turning it on. 

If it's passing all the plugins it's probably ok to just go ahead with, and seems like a really great thing for our users though.  WDYT?

Reply:
Discussed offline